### ISSUE TITLE
Define "definition", "node", and "item" more precisely in section on hir

### ISSUE THREAD

**Issue by rylev** [rust-lang/rustc-dev-guide#1009]
In the guide on [hir](https://rustc-dev-guide.rust-lang.org/hir.html), the terms "definition", "node", and "item" are used without much explanation which leaves the terms vague and fuzzy. If possible, we should try to more precisely define them so that newcomers have a better picture in their head of what they refer to.

**Comment by jyn514** [comment]
- [`Node`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.Node.html)
- [`Item`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.Item.html) (but I haven't read the text, it might also mean the general concept of an item: https://doc.rust-lang.org/stable/reference/items.html#items)
- Maybe 'definition' is https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/definitions/struct.Definitions.html ?

**Comment by LeSeulArtichaut** [comment]
I think "definition" is anything that has a DefId, which can be any variant of [DefKind](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def/enum.DefKind.html)

**Comment by rylev** [comment]
I don't think this is fully statisfying of an answer. I think we can certainly say that the precise answers are what was linked above, but I think it's also important to give approximate answers. For instance, the definition of `definition` could be:
>  Loosely, anything in Rust where something is defined (as opposed to used). For example, a struct, an enum, a trait, etc.). More precisely, anything with an associated `DefId`.
The goal is provide an intuition of what these things are not to come up with the absolutely most precise definition we can think of.

**Comment by camelid** [comment]
*That's a bit of a circular definition though: *a definition is something that's defined.

**Comment by rylev** [comment]
I acknowledge it’s a bit circular, but it can be clarifying particularly if we add “defined in Rust code” as that can clarify to people we're talking about a direct definition from code and not something more specific. 

FWIW, if I had seen this when first reading the guide, the explanation would have been clearer to me.
